src/main/kotlin/org/rust/ide/refactoring/RsRenameProcessor.kt

```diff
@@ -126,12 +132,42 @@ class RsRenameProcessor : RenamePsiElementProcessor() {
         allRenames[element] = rename
     }
 
-    override fun substituteElementToRename(element: PsiElement, editor: Editor?): PsiElement =
-        (element as? RsAbstractable)?.superItem ?: element
+    override fun substituteElementToRename(element: PsiElement, editor: Editor?): PsiElement {
+        val superElement = (element as? RsAbstractable)?.superItem ?: element
+        return superElement.findFakeElementForRenameInMacroBody() ?: superElement
+    }
+
+    private fun PsiElement.findFakeElementForRenameInMacroBody(): PsiElement? {
+        if (this is RsNameIdentifierOwner) {
+            val identifier = nameIdentifier
+            val sourceIdentifier = identifier?.findElementExpandedFrom()
+            if (sourceIdentifier != null) {
+                when (val sourceIdentifierParent = sourceIdentifier.parent) {
```

Comment:
Is it possible to handle case when `sourceIdentifierParent` is `RsPath`? Like in #8182, when renaming parameter of attr proc macro:

```rust
#[attr(/*caret*/Foo)]
fn func() { ... }
```